### PR TITLE
Feat/728 new version check

### PR DESCRIPTION
**Pull request by rick23p**

### CODE REVIEW

File: src/locales/pt_br/translation.json [hunk 1]
```diff
@@ -180,6 +180,12 @@
   "hintDialog": {
     "title": "Dicas Sovryn"
   },
+  "updateDaapDialog": {
+    "title": "Sovryn foi atualizado para <br/> <0>Build ID: {{buildId}}</0>",
```
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:184]:
```suggestion
    "title": "A Sovryn foi atualizada para a versão <br/> <0>Build ID: {{buildId}}</0>",
```

File: src/locales/pt_br/translation.json [hunk 2]
```diff
@@ -180,6 +180,12 @@
...
+    "notice": "Feche todas as guias com o Sovryn aberto para atualizar",
```
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:185]:
```suggestion
    "notice": "Para atualizar, feche a página da Sovryn ",
```

File: src/locales/pt_br/translation.json [hunk 3]
```diff
@@ -180,6 +180,12 @@
...
+    "updateBtn": "ATUALIZAR",
+    "closeBtn": "Por favor, deixe-me fechar uma negociação"
```
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:187]:
```suggestion
    "closeBtn": "Por favor, deixe-me finalizar uma negociação"
```

**Review by bitcoinuser — CHANGES_REQUESTED** [review]
Changes PT-BR translation

**Review by creed-victor — CHANGES_REQUESTED** [review]
This should be same design as the modal you did.

Screenshot 2021-04-20 at 16 50 33

Actually, your modal should be moved to this component and just have this extra functionality of checking updates with timer.

Component name:  `ServiceWorkerToaster`

**Comment by rick23p** [comment]
> This should be same design as the modal you did.
> 
> Screenshot 2021-04-20 at 16 50 33
> 
> Actually, your modal should be moved to this component and just have this extra functionality of checking updates with timer.
> 
> Component name: `ServiceWorkerToaster`

Resolved

File: src/app/components/ServiceWorkerToaster/index.tsx [hunk 4]
```diff
@@ -3,58 +3,112 @@
...
+import styles from './index.module.css';
+import logoSvg from 'assets/images/sovryn-logo-horz-white.png';
+import { Button } from '../Button';
+import axios from 'axios';
+import { soliditySha3 } from 'web3-utils';
 
 interface Props {}
 
+//interval time to check sw
+const CHECK_TIME = 10 * 60 * 1000;
```
Comment (creed-victor) [inline src/app/components/ServiceWorkerToaster/index.tsx:21]:
```suggestion
const CHECK_TIME = 15e3; // 15 seconds
```

File: src/app/components/ServiceWorkerToaster/index.tsx [hunk 5]
```diff
@@ -3,58 +3,112 @@
...
+            onClick={() => setShow(false)}
+          />
+        </div>
+      </div>
+    </Dialog>
+  );
 }
+
+const StyledLogo = styled.img.attrs(_ => ({
+  alt: 'logo',
```
Comment (creed-victor) [inline src/app/components/ServiceWorkerToaster/index.tsx:111]: Just a dummy change to force redeployment.
```suggestion
  alt: 'Sovryn',
```

**Review by creed-victor — APPROVED** [review]

File: src/index.tsx [hunk 6]
```diff
@@ -49,6 +49,7 @@ const ConnectedApp = ({ Component }: Props) => {
         {/*</React.StrictMode>*/}
       </HelmetProvider>
       <ServiceWorkerToaster />
+
```
Comment (creed-victor) [inline src/index.tsx:52]:
```suggestion
```

File: src/app/components/ServiceWorkerToaster/index.module.css [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  background-color: #2274a5bf !important;
+  border: 9px solid #2274A5;
+  border-radius: 20px;
+  opacity: 1;
+}
+.close {
+  font-size: 10px;
+  height: 25px;
+  border-radius: 5px;
+}
```
Comment (creed-victor) [inline src/app/components/ServiceWorkerToaster/index.module.css:11]:
```suggestion
}

```

**Review by creed-victor — APPROVED** [review]

File: src/app/components/ServiceWorkerToaster/index.tsx [hunk 8]
```diff
@@ -3,58 +3,144 @@
...
+        if (first) setOldSW(hash || '');
+        else setNewSW(hash || '');
+      })
+      .catch(() => {});
+  };
+
+  useEffect(() => {
+    if (!oldSW && newSW) setOldSW(newSW);
+    if (oldSW && newSW && oldSW !== newSW) setShow(true);
+
```
Comment (creed-victor) [inline src/app/components/ServiceWorkerToaster/index.tsx:71]:
```suggestion
```

**Review by t-z-u — APPROVED** [review]

**Pull request merged**